src/web/demo/src/main/resources/org/geoserver/web/demo/demo-requests.js

```diff
@@ -0,0 +1,283 @@
+// case 1 - image (convert to base64 Image representation and display)
+// case 2 - other (XML - display the XML - might be pretty printed)
+function handleResponse(xhrResponse) {
+    var contentType = xhrResponse.getResponseHeader("Content-Type");
+    var status = xhrResponse.status;
+
+    //decode binary -> text
+    var enc = new TextDecoder("utf-8");
+    var text = enc.decode(xhrResponse.response);
+
+    if (contentType.startsWith("image") && text.startsWith("<?xml version=\"1.0\" encoding=\"UTF-8\"?>")) {
+        //this is actually a service exception that is tagged as an image
+        contentType = "text/xml";
+    }
+    displayHeaders(xhrResponse);
+
+    //error status
+    if (status > 400) {
+        document.getElementById("responseDiv").style.borderColor = 'red';
+    }
+    else {
+        document.getElementById("responseDiv").style.borderColor = 'black';
+    }
+    //its an image - convert to Base64 representation and display in an <img> tag
+    if (contentType.startsWith("image")) {
+        //var base64Image = btoa(String.fromCharCode.apply(null, new Uint8Array(xhrResponse.response)));
+        var base64Image = arrayBufferToBase64(xhrResponse.response);
+
+        var responseDiv =    document.getElementById("responseDiv");
+        //remove old content from the responseDiv
+        while(responseDiv.firstChild){
+            responseDiv.removeChild(responseDiv.firstChild);
+        }
+
+        //create the <img> and use the base64 image data
+        let imgElement = document.createElement("img");
+        imgElement.src = "data:"+contentType+";base64, "+base64Image;
+        responseDiv.append(imgElement);
+    }
+    else {
+        //its probably XML - process it for display in a <div><pre>...</pre><div>
+        var responseDiv =    document.getElementById("responseDiv");
+        //remove old content from the responseDiv
+        while(responseDiv.firstChild){
+            responseDiv.removeChild(responseDiv.firstChild);
+        }
+
+        //pretty print XML if its checked
+        if (document.getElementById("prettyXML").checked) {
+            const options = {indent: 2, newline: '\n'}
+            text = prettifyXml(text, options);
+        }
+
+        //replace HTML/XML characters for display (and security)
+        text=text.replaceAll("<","&lt;");
+        text=text.replaceAll(">","&gt;");
+
+        //create the <pre> element and imbed inside the <div>
+        let preElement = document.createElement("pre");
+        preElement.innerHTML = text;
+        responseDiv.append(preElement);
+    }
+}
+
+function arrayBufferToBase64(buffer) {
+    let binary = '';
+    const bytes = new Uint8Array(buffer);
+    for (let i = 0; i < bytes.byteLength; i++) {
+        binary += String.fromCharCode(bytes[i]);
+    }
+    return btoa(binary);
+}
+
+// ======================================================
+// https://www.npmjs.com/package/prettify-xml
+//
+// The MIT License (MIT)
```

Comment:
There is an MIT License right in the middle of the javascript, is that intended?

Reply:
Yes - there's MIT code in the JS.  Thats the "xml pretty print" code.

I can move it to its own file??